// @aws-sdk/client-bedrock-data-automation/dist-es/models/models_0.js
import { SENSITIVE_STRING } from "@smithy/smithy-client";
import { BedrockDataAutomationServiceException as __BaseException } from "./BedrockDataAutomationServiceException";
export class AccessDeniedException extends __BaseException {
    name = "AccessDeniedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
    }
}
export class ConflictException extends __BaseException {
    name = "ConflictException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
    }
}
export const BlueprintStage = {
    DEVELOPMENT: "DEVELOPMENT",
    LIVE: "LIVE",
};
export const Type = {
    DOCUMENT: "DOCUMENT",
    IMAGE: "IMAGE",
};
export class InternalServerException extends __BaseException {
    name = "InternalServerException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServerException.prototype);
    }
}
export class ServiceQuotaExceededException extends __BaseException {
    name = "ServiceQuotaExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
    }
}
export class ThrottlingException extends __BaseException {
    name = "ThrottlingException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ThrottlingException.prototype);
    }
}
export class ValidationException extends __BaseException {
    name = "ValidationException";
    $fault = "client";
    fieldList;
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.fieldList = opts.fieldList;
    }
}
export class ResourceNotFoundException extends __BaseException {
    name = "ResourceNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
    }
}
export const BlueprintStageFilter = {
    ALL: "ALL",
    DEVELOPMENT: "DEVELOPMENT",
    LIVE: "LIVE",
};
export const DataAutomationProjectStage = {
    DEVELOPMENT: "DEVELOPMENT",
    LIVE: "LIVE",
};
export const ResourceOwner = {
    ACCOUNT: "ACCOUNT",
    SERVICE: "SERVICE",
};
export const State = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const AudioExtractionCategoryType = {
    AUDIO_CONTENT_MODERATION: "AUDIO_CONTENT_MODERATION",
    TOPIC_CONTENT_MODERATION: "TOPIC_CONTENT_MODERATION",
    TRANSCRIPT: "TRANSCRIPT",
};
export const AudioStandardGenerativeFieldType = {
    AUDIO_SUMMARY: "AUDIO_SUMMARY",
    IAB: "IAB",
    TOPIC_SUMMARY: "TOPIC_SUMMARY",
};
export const DocumentExtractionGranularityType = {
    DOCUMENT: "DOCUMENT",
    ELEMENT: "ELEMENT",
    LINE: "LINE",
    PAGE: "PAGE",
    WORD: "WORD",
};
export const DocumentOutputTextFormatType = {
    CSV: "CSV",
    HTML: "HTML",
    MARKDOWN: "MARKDOWN",
    PLAIN_TEXT: "PLAIN_TEXT",
};
export const ImageExtractionCategoryType = {
    CONTENT_MODERATION: "CONTENT_MODERATION",
    LOGOS: "LOGOS",
    TEXT_DETECTION: "TEXT_DETECTION",
};
export const ImageStandardGenerativeFieldType = {
    IAB: "IAB",
    IMAGE_SUMMARY: "IMAGE_SUMMARY",
};
export const VideoExtractionCategoryType = {
    CONTENT_MODERATION: "CONTENT_MODERATION",
    LOGOS: "LOGOS",
    TEXT_DETECTION: "TEXT_DETECTION",
    TRANSCRIPT: "TRANSCRIPT",
};
export const VideoStandardGenerativeFieldType = {
    CHAPTER_SUMMARY: "CHAPTER_SUMMARY",
    IAB: "IAB",
    VIDEO_SUMMARY: "VIDEO_SUMMARY",
};
export const DataAutomationProjectStatus = {
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
};
export const DataAutomationProjectStageFilter = {
    ALL: "ALL",
    DEVELOPMENT: "DEVELOPMENT",
    LIVE: "LIVE",
};
export const CreateBlueprintRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.blueprintName && { blueprintName: SENSITIVE_STRING }),
    ...(obj.schema && { schema: SENSITIVE_STRING }),
});
export const BlueprintFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.schema && { schema: SENSITIVE_STRING }),
    ...(obj.blueprintName && { blueprintName: SENSITIVE_STRING }),
});
export const CreateBlueprintResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.blueprint && { blueprint: BlueprintFilterSensitiveLog(obj.blueprint) }),
});
export const GetBlueprintResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.blueprint && { blueprint: BlueprintFilterSensitiveLog(obj.blueprint) }),
});
export const BlueprintSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.blueprintName && { blueprintName: SENSITIVE_STRING }),
});
export const ListBlueprintsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.blueprints && { blueprints: obj.blueprints.map((item) => BlueprintSummaryFilterSensitiveLog(item)) }),
});
export const UpdateBlueprintRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.schema && { schema: SENSITIVE_STRING }),
});
export const UpdateBlueprintResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.blueprint && { blueprint: BlueprintFilterSensitiveLog(obj.blueprint) }),
});
export const CreateBlueprintVersionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.blueprint && { blueprint: BlueprintFilterSensitiveLog(obj.blueprint) }),
});
export const CreateDataAutomationProjectRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.projectName && { projectName: SENSITIVE_STRING }),
    ...(obj.projectDescription && { projectDescription: SENSITIVE_STRING }),
});
export const DataAutomationProjectFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.projectName && { projectName: SENSITIVE_STRING }),
    ...(obj.projectDescription && { projectDescription: SENSITIVE_STRING }),
});
export const GetDataAutomationProjectResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.project && { project: DataAutomationProjectFilterSensitiveLog(obj.project) }),
});
export const DataAutomationProjectSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.projectName && { projectName: SENSITIVE_STRING }),
});
export const ListDataAutomationProjectsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.projects && { projects: obj.projects.map((item) => DataAutomationProjectSummaryFilterSensitiveLog(item)) }),
});
export const UpdateDataAutomationProjectRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.projectDescription && { projectDescription: SENSITIVE_STRING }),
});
