// @aws-sdk/client-bedrock-data-automation/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateDataAutomationProjectRequestFilterSensitiveLog = exports.ListDataAutomationProjectsResponseFilterSensitiveLog = exports.DataAutomationProjectSummaryFilterSensitiveLog = exports.GetDataAutomationProjectResponseFilterSensitiveLog = exports.DataAutomationProjectFilterSensitiveLog = exports.CreateDataAutomationProjectRequestFilterSensitiveLog = exports.CreateBlueprintVersionResponseFilterSensitiveLog = exports.UpdateBlueprintResponseFilterSensitiveLog = exports.UpdateBlueprintRequestFilterSensitiveLog = exports.ListBlueprintsResponseFilterSensitiveLog = exports.BlueprintSummaryFilterSensitiveLog = exports.GetBlueprintResponseFilterSensitiveLog = exports.CreateBlueprintResponseFilterSensitiveLog = exports.BlueprintFilterSensitiveLog = exports.CreateBlueprintRequestFilterSensitiveLog = exports.DataAutomationProjectStageFilter = exports.DataAutomationProjectStatus = exports.VideoStandardGenerativeFieldType = exports.VideoExtractionCategoryType = exports.ImageStandardGenerativeFieldType = exports.ImageExtractionCategoryType = exports.DocumentOutputTextFormatType = exports.DocumentExtractionGranularityType = exports.AudioStandardGenerativeFieldType = exports.AudioExtractionCategoryType = exports.State = exports.ResourceOwner = exports.DataAutomationProjectStage = exports.BlueprintStageFilter = exports.ResourceNotFoundException = exports.ValidationException = exports.ThrottlingException = exports.ServiceQuotaExceededException = exports.InternalServerException = exports.Type = exports.BlueprintStage = exports.ConflictException = exports.AccessDeniedException = void 0;
const smithy_client_1 = require("@smithy/smithy-client");
const BedrockDataAutomationServiceException_1 = require("./BedrockDataAutomationServiceException");
class AccessDeniedException extends BedrockDataAutomationServiceException_1.BedrockDataAutomationServiceException {
    name = "AccessDeniedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
    }
}
exports.AccessDeniedException = AccessDeniedException;
class ConflictException extends BedrockDataAutomationServiceException_1.BedrockDataAutomationServiceException {
    name = "ConflictException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
    }
}
exports.ConflictException = ConflictException;
exports.BlueprintStage = {
    DEVELOPMENT: "DEVELOPMENT",
    LIVE: "LIVE",
};
exports.Type = {
    DOCUMENT: "DOCUMENT",
    IMAGE: "IMAGE",
};
class InternalServerException extends BedrockDataAutomationServiceException_1.BedrockDataAutomationServiceException {
    name = "InternalServerException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServerException.prototype);
    }
}
exports.InternalServerException = InternalServerException;
class ServiceQuotaExceededException extends BedrockDataAutomationServiceException_1.BedrockDataAutomationServiceException {
    name = "ServiceQuotaExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
    }
}
exports.ServiceQuotaExceededException = ServiceQuotaExceededException;
class ThrottlingException extends BedrockDataAutomationServiceException_1.BedrockDataAutomationServiceException {
    name = "ThrottlingException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ThrottlingException.prototype);
    }
}
exports.ThrottlingException = ThrottlingException;
class ValidationException extends BedrockDataAutomationServiceException_1.BedrockDataAutomationServiceException {
    name = "ValidationException";
    $fault = "client";
    fieldList;
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.fieldList = opts.fieldList;
    }
}
exports.ValidationException = ValidationException;
class ResourceNotFoundException extends BedrockDataAutomationServiceException_1.BedrockDataAutomationServiceException {
    name = "ResourceNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
    }
}
exports.ResourceNotFoundException = ResourceNotFoundException;
exports.BlueprintStageFilter = {
    ALL: "ALL",
    DEVELOPMENT: "DEVELOPMENT",
    LIVE: "LIVE",
};
exports.DataAutomationProjectStage = {
    DEVELOPMENT: "DEVELOPMENT",
    LIVE: "LIVE",
};
exports.ResourceOwner = {
    ACCOUNT: "ACCOUNT",
    SERVICE: "SERVICE",
};
exports.State = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
exports.AudioExtractionCategoryType = {
    AUDIO_CONTENT_MODERATION: "AUDIO_CONTENT_MODERATION",
    TOPIC_CONTENT_MODERATION: "TOPIC_CONTENT_MODERATION",
    TRANSCRIPT: "TRANSCRIPT",
};
exports.AudioStandardGenerativeFieldType = {
    AUDIO_SUMMARY: "AUDIO_SUMMARY",
    IAB: "IAB",
    TOPIC_SUMMARY: "TOPIC_SUMMARY",
};
exports.DocumentExtractionGranularityType = {
    DOCUMENT: "DOCUMENT",
    ELEMENT: "ELEMENT",
    LINE: "LINE",
    PAGE: "PAGE",
    WORD: "WORD",
};
exports.DocumentOutputTextFormatType = {
    CSV: "CSV",
    HTML: "HTML",
    MARKDOWN: "MARKDOWN",
    PLAIN_TEXT: "PLAIN_TEXT",
};
exports.ImageExtractionCategoryType = {
    CONTENT_MODERATION: "CONTENT_MODERATION",
    LOGOS: "LOGOS",
    TEXT_DETECTION: "TEXT_DETECTION",
};
exports.ImageStandardGenerativeFieldType = {
    IAB: "IAB",
    IMAGE_SUMMARY: "IMAGE_SUMMARY",
};
exports.VideoExtractionCategoryType = {
    CONTENT_MODERATION: "CONTENT_MODERATION",
    LOGOS: "LOGOS",
    TEXT_DETECTION: "TEXT_DETECTION",
    TRANSCRIPT: "TRANSCRIPT",
};
exports.VideoStandardGenerativeFieldType = {
    CHAPTER_SUMMARY: "CHAPTER_SUMMARY",
    IAB: "IAB",
    VIDEO_SUMMARY: "VIDEO_SUMMARY",
};
exports.DataAutomationProjectStatus = {
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
};
exports.DataAutomationProjectStageFilter = {
    ALL: "ALL",
    DEVELOPMENT: "DEVELOPMENT",
    LIVE: "LIVE",
};
const CreateBlueprintRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.blueprintName && { blueprintName: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.schema && { schema: smithy_client_1.SENSITIVE_STRING }),
});
exports.CreateBlueprintRequestFilterSensitiveLog = CreateBlueprintRequestFilterSensitiveLog;
const BlueprintFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.schema && { schema: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.blueprintName && { blueprintName: smithy_client_1.SENSITIVE_STRING }),
});
exports.BlueprintFilterSensitiveLog = BlueprintFilterSensitiveLog;
const CreateBlueprintResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.blueprint && { blueprint: (0, exports.BlueprintFilterSensitiveLog)(obj.blueprint) }),
});
exports.CreateBlueprintResponseFilterSensitiveLog = CreateBlueprintResponseFilterSensitiveLog;
const GetBlueprintResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.blueprint && { blueprint: (0, exports.BlueprintFilterSensitiveLog)(obj.blueprint) }),
});
exports.GetBlueprintResponseFilterSensitiveLog = GetBlueprintResponseFilterSensitiveLog;
const BlueprintSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.blueprintName && { blueprintName: smithy_client_1.SENSITIVE_STRING }),
});
exports.BlueprintSummaryFilterSensitiveLog = BlueprintSummaryFilterSensitiveLog;
const ListBlueprintsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.blueprints && { blueprints: obj.blueprints.map((item) => (0, exports.BlueprintSummaryFilterSensitiveLog)(item)) }),
});
exports.ListBlueprintsResponseFilterSensitiveLog = ListBlueprintsResponseFilterSensitiveLog;
const UpdateBlueprintRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.schema && { schema: smithy_client_1.SENSITIVE_STRING }),
});
exports.UpdateBlueprintRequestFilterSensitiveLog = UpdateBlueprintRequestFilterSensitiveLog;
const UpdateBlueprintResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.blueprint && { blueprint: (0, exports.BlueprintFilterSensitiveLog)(obj.blueprint) }),
});
exports.UpdateBlueprintResponseFilterSensitiveLog = UpdateBlueprintResponseFilterSensitiveLog;
const CreateBlueprintVersionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.blueprint && { blueprint: (0, exports.BlueprintFilterSensitiveLog)(obj.blueprint) }),
});
exports.CreateBlueprintVersionResponseFilterSensitiveLog = CreateBlueprintVersionResponseFilterSensitiveLog;
const CreateDataAutomationProjectRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.projectName && { projectName: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.projectDescription && { projectDescription: smithy_client_1.SENSITIVE_STRING }),
});
exports.CreateDataAutomationProjectRequestFilterSensitiveLog = CreateDataAutomationProjectRequestFilterSensitiveLog;
const DataAutomationProjectFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.projectName && { projectName: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.projectDescription && { projectDescription: smithy_client_1.SENSITIVE_STRING }),
});
exports.DataAutomationProjectFilterSensitiveLog = DataAutomationProjectFilterSensitiveLog;
const GetDataAutomationProjectResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.project && { project: (0, exports.DataAutomationProjectFilterSensitiveLog)(obj.project) }),
});
exports.GetDataAutomationProjectResponseFilterSensitiveLog = GetDataAutomationProjectResponseFilterSensitiveLog;
const DataAutomationProjectSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.projectName && { projectName: smithy_client_1.SENSITIVE_STRING }),
});
exports.DataAutomationProjectSummaryFilterSensitiveLog = DataAutomationProjectSummaryFilterSensitiveLog;
const ListDataAutomationProjectsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.projects && { projects: obj.projects.map((item) => (0, exports.DataAutomationProjectSummaryFilterSensitiveLog)(item)) }),
});
exports.ListDataAutomationProjectsResponseFilterSensitiveLog = ListDataAutomationProjectsResponseFilterSensitiveLog;
const UpdateDataAutomationProjectRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.projectDescription && { projectDescription: smithy_client_1.SENSITIVE_STRING }),
});
exports.UpdateDataAutomationProjectRequestFilterSensitiveLog = UpdateDataAutomationProjectRequestFilterSensitiveLog;
