// @aws-sdk/client-bedrock-data-automation-runtime/dist-cjs/BedrockDataAutomationRuntime.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BedrockDataAutomationRuntime = void 0;
const smithy_client_1 = require("@smithy/smithy-client");
const BedrockDataAutomationRuntimeClient_1 = require("./BedrockDataAutomationRuntimeClient");
const GetDataAutomationStatusCommand_1 = require("./commands/GetDataAutomationStatusCommand");
const InvokeDataAutomationAsyncCommand_1 = require("./commands/InvokeDataAutomationAsyncCommand");
const ListTagsForResourceCommand_1 = require("./commands/ListTagsForResourceCommand");
const TagResourceCommand_1 = require("./commands/TagResourceCommand");
const UntagResourceCommand_1 = require("./commands/UntagResourceCommand");
const commands = {
    GetDataAutomationStatusCommand: GetDataAutomationStatusCommand_1.GetDataAutomationStatusCommand,
    InvokeDataAutomationAsyncCommand: InvokeDataAutomationAsyncCommand_1.InvokeDataAutomationAsyncCommand,
    ListTagsForResourceCommand: ListTagsForResourceCommand_1.ListTagsForResourceCommand,
    TagResourceCommand: TagResourceCommand_1.TagResourceCommand,
    UntagResourceCommand: UntagResourceCommand_1.UntagResourceCommand,
};
class BedrockDataAutomationRuntime extends BedrockDataAutomationRuntimeClient_1.BedrockDataAutomationRuntimeClient {
}
exports.BedrockDataAutomationRuntime = BedrockDataAutomationRuntime;
(0, smithy_client_1.createAggregatedClient)(commands, BedrockDataAutomationRuntime);
